### PR TITLE
add dynamic support for embedding_bag/index_select

### PR DESCRIPTION
**Pull request by lanluo-nvidia**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: tests/py/dynamo/conversion/test_embedding_bag_aten.py [hunk 1]
```diff
@@ -408,6 +410,86 @@ def forward(self, weight, indices, offsets):
...
+        if per_sample_weights is not None:
+            dynamic_shapes["per_sample_weights"] = {}
+        fx_mod = torch.export.export(mod, inputs, dynamic_shapes=dynamic_shapes)
+        trt_mod = torch_tensorrt.dynamo.compile(
+            fx_mod, inputs=inputs, enable_precisions=torch.float32, min_block_size=1
+        )
+
+        with torch.no_grad():
+            cuda_inputs = []
+            for i in inputs:
```
Comment (peri044) [inline tests/py/dynamo/conversion/test_embedding_bag_aten.py:479]: Instead of using the same inputs that were used to compile, can you try with another shape within the dynamic shape range ? Like in the first example, it would be (6, 2)

File: tests/py/dynamo/conversion/test_index_select_aten.py [hunk 2]
```diff
@@ -36,6 +38,86 @@ def forward(self, source_tensor, indices_tensor):
...
+        mod = IndexSelect()
+
+        fx_mod = torch.export.export(mod, inputs, dynamic_shapes=dynamic_shapes)
+        trt_mod = torch_tensorrt.dynamo.compile(
+            fx_mod, inputs=inputs, enable_precisions=torch.float32, min_block_size=1
+        )
+
+        with torch.no_grad():
+            cuda_inputs = []
+            for i in inputs:
```
Comment (peri044) [inline tests/py/dynamo/conversion/test_index_select_aten.py:107]: Same comment as above

**Review by peri044 — APPROVED** [review]
LGTM
